### ISSUE TITLE
4.0.0-rc.9: removeEventListener error from ion-slides when unloading from a nested ion-router-outlet

### ISSUE THREAD

**Issue by TehCrucible** [ionic-team/ionic#14780]
# Bug Report

**Ionic Info**
```
Ionic:

   ionic (Ionic CLI)          : 4.0.0-rc.9 (C:\Users\jamie\AppData\Roaming\nvm\v8.9.4\node_modules\ionic)
   Ionic Framework            : @ionic/angular 4.0.0-alpha.9
   @angular-devkit/core       : 0.7.0-rc.1
   @angular-devkit/schematics : 0.7.0-rc.1
   @angular/cli               : 6.0.8
   @ionic/ng-toolkit          : 1.0.0-rc.10
   @ionic/schematics-angular  : 1.0.0-rc.10

System:

   NodeJS : v8.9.4 (C:\Program Files\nodejs\node.exe)
   npm    : 5.6.0
   OS     : Windows 10
```

**Describe the Bug**
When using a nested `ion-router-outlet` (or a regular `router-outlet` for that matter), you can't unload a component that contains an `ion-slides` components without causing the following error:

```
Uncaught TypeError: Cannot read property 'removeEventListener' of undefined
    at t.disableDraggable (4g52razy.js:5)
    at t.destroy (4g52razy.js:5)
    at t.destroy (4g52razy.js:5)
    at 4g52razy.js:5
    at Array.forEach (<anonymous>)
    at 4g52razy.js:5
    at Array.forEach (<anonymous>)
    at t.push../node_modules/@ionic/core/dist/esm/es5/4g52razy.js.h.emit (4g52razy.js:5)
    at t.destroy (4g52razy.js:5)
    at e.componentDidUnload (4g52razy.js:5)
```

**Steps to Reproduce**
1.  Create a page/module with child routes and an ion-router-outlet
2.  Create a component with an ion-slides
3.  Load and unload the component using routing

**Related Code**

home.page.html
```
<ion-content>
  <a routerLink="/home/first">First</a>
  <a routerLink="/home/second">Second</a>
  <router-outlet></router-outlet>
</ion-content>
```

home.module.ts
```
import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";
import { FormsModule } from "@angular/forms";
import { RouterModule } from "@angular/router";
import { IonicModule } from "@ionic/angular";

import { ComponentsModule } from "../../components/components";
import { FirstComponent } from "../../components/first/first.component";
import { SecondComponent } from "../../components/second/second.component";
import { HomePage } from "./home.page";

@NgModule({
  imports: [
    CommonModule,
    FormsModule,
    IonicModule,
    ComponentsModule,
    RouterModule.forChild([
      {
        path: '',
        component: HomePage,
        children: [
          { path: 'first', component: FirstComponent },
          { path: 'second', component: SecondComponent },
        ]
      }
    ])
  ],
  declarations: [HomePage]
})
export class HomePageModule { }
```

first.component.html
```
<ion-slides>
  <ion-slide>
    First
  </ion-slide>
</ion-slides>
```

**Expected Behavior**
Unload the ion-slides component without causing an error.

**Comment by thefont** [comment]
I'm having this problem as well. It seems like the problem exists whenever the component is destroyed, though. I was able to recreate the error by wrapping the ion-slides component in a div and adding an *ngIf directive that would evaluate to false after a button was pressed.

example.html
```
<ion-content>
    <div *ngIf="!destroy">
        <ion-slides #images pager="true">
            <ion-slide *ngFor="let url of imageUrls">
                <div>
                    <img [src]="url" />
                </div>
            </ion-slide>
        </ion-slides>
    </div>
    <ion-button (click)="toggleSlides()">Toggle Slides</ion-button>
</ion-content>
```

example.component.ts
```
@Component({
    selector: 'example',
    templateUrl: 'example.component.html'
})
export class ExampleComponent implements OnInit {
    private destroy = false;
    private imageUrls: Array<string> = [
        'https://farm4.staticflickr.com/3335/4615871733_f5aaab9400_b.jpg',
    ]
    
    constructor() { }

    ngOnInit() { }

    private toggleSlides() {
        this.destroy = !this.destroy;
    }
}
```

**Comment by jchapelle** [comment]
+1

**Comment by clarencecastillo** [comment]
I believe this was already fixed in 4.0.0-beta.1. Check out https://github.com/ionic-team/ionic/pull/14865.
